src/bin/pgcopydb/pgsql.c

```diff
@@ -2273,13 +2270,37 @@ pg_copy(PGSQL *src, PGSQL *dst, const char *srcQname, const char *dstQname,
 		return false;
 	}
 
+	bool result = pg_copy_data(src, dst, srcQname, dstQname, truncate);
+
+	if (srcConnIsOurs)
+	{
+		pgsql_finish(src);
+	}
+
+	if (dstConnIsOurs)
+	{
+		pgsql_finish(dst);
+	}
+
+	return result;
+}
+
+
+/*
+ * pg_copy implements a COPY operation from a source Postgres instance (src) to
+ * a target Postgres instance (dst), for the data found in the table referenced
+ * by the qualified identifier name srcQname on the source, into the table
+ * referenced by the qualified identifier name dstQname on the target.
+ */
+bool
+pg_copy_data(PGSQL *src, PGSQL *dst, const char *srcQname, const char *dstQname,
+			 bool truncate)
```

Comment:
Here I have extracted out the function from `pg_copy` to make the setting up and clean-up of connection from failure simpler. 
It's sort of a private function but since C doesn't have such construct is there any convention to show it's not meant for other files to use.

Reply:
Make it static, you will find other functions in there that are static. Then it needs its prototype in the beginning of the same .c file.